src/devices/Mcp9808/Mcp9808.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Units;
+
+namespace Iot.Device.Mcp9808
+{
+    /// <summary>
+    /// Microchip's MCP9808 I2C Temp sensor
+    /// </summary>
+    public class Mcp9808 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// MCP9808 I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x18;
+
+        #region prop
+
+        /// <summary>
+        /// MCP9808 Temperature
+        /// </summary>
+        public Temperature Temperature { get => Temperature.FromCelsius(GetTemperature()); }
+
+        private bool _disable;
+
+        /// <summary>
+        /// Disable MCP9808
+        /// </summary>
+        public bool Disabled
+        {
+            get
+            {
+                return _disable;
+            }
+            set
+            {
+                SetShutdown(value);
```

Comment:
NIT: should we check here `if (value != _disable)` so that we only do it when we are changing the value?

Reply:
I don't think there's any harm in setting the registry twice, maybe for some reason you'd want to re-send the reg-write?

Reply:
IMHO these classes are fairly low-level and shouldn't have too much logic/"optimization", as long as it doesn't cause a fault-state within the device if the registry is written to multiple times.